### PR TITLE
[Xamarin.Android.Build.Tests] Import tests from last regression test job 

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/QualityAssurance/pull/3646

CI and Xamarin.Android.Build.Tests have been updated to reflect the
third round of regression test auditing. The Xamarin.Android YAML
pipeline no longer contains QA test jobs, and the following regression
test categories have been removed:

 * `RuntimeTests` - Still runs on multiple devices on the [external test job][1].
 * `Hello` - Still runs on multiple devices on the [external test job][1].
 * `RegressionDeviceTests` - Obsolete tests removed, some missing coverage migrated.
 * `FastDevDeviceTests` - Obsolete tests removed, some missing coverage migrated.
 * `BuildPerformance` - Missing coverage migrated.
 * `AotSupport` - Obsolete tests removed, some missing coverage migrated.

The `dotnet` test categories have been removed as they are no longer
used as far as I can tell.  The `DotNetIgnore` category is the preferred
way to filter tests that aren't compatible with One .NET.

`AndroidUpdateResourceTest.RepetitiveBuild` has been removed, as it
appeared to be a duplicate of
`IncrementalBuildTest.BasicApplicationRepetitiveBuild`.
`BuildTest.BuildReleaseApplication` has also been removed as it is
implicitly covered by other release configuration tests.

[1]: https://dev.azure.com/devdiv/DevDiv/_release?definitionId=476&view=mine&_a=releases.

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AotTests.cs [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Mono.Cecil;
+using NUnit.Framework;
+using Xamarin.ProjectTools;
+
+namespace Xamarin.Android.Build.Tests
+{
+	[Category ("Node-1"), Category ("AOT")]
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AotTests.cs:13]: I think `Node-1` is already taking the longest, should we put these in `Node-2` or `Node-3`?

Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AotTests.cs:13]: Good idea 👍, I was going to wait to see how timings looked after these changes run and try to do some rebalancing.

Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/AotTests.cs]: `Node-2` seems most preferable here, the latest PR build shows the nunit3 test step on these legacy jobs taking anywhere from 30-43 minutes on Windows and 42-54 minutes on macOS.

**Comment by pjcollins** [comment]
Full test run queued: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=4063251&view=results

**Review by jonathanpeppers — APPROVED** [review]
This looks good, if the build ends up green. 👍

**Pull request merged**